src/main/java/frc/robot/subsystems/Shooter.java

```diff
@@ -77,11 +77,14 @@ public void stopShooter() {
   public boolean isReadyToShoot() {
     double rightMotorVelocity = m_rightEncoder.getVelocity();
     double leftMotorVelocity = m_leftEncoder.getVelocity();
+
+    double rightTolerance = m_idealRightSpeed * m_kShootingTolerance;
+    double leftTolerance = m_idealRightSpeed * m_kShootingTolerance;
     return (
-      rightMotorVelocity <= m_idealRightSpeed + m_kShootingTolerance || 
-      rightMotorVelocity >= m_idealRightSpeed - m_kShootingTolerance &&
-      leftMotorVelocity <= m_idealLeftSpeed + m_kShootingTolerance || 
-      leftMotorVelocity >= m_idealLeftSpeed - m_kShootingTolerance
+      rightMotorVelocity * (1.0 - rightTolerance) <= m_idealRightSpeed || 
+      rightMotorVelocity * (1.0 + rightTolerance) >= m_idealRightSpeed &&
+      leftMotorVelocity  * (1.0 -leftTolerance) <= m_idealLeftSpeed || 
+      leftMotorVelocity  * (1.0 + leftTolerance) >= m_idealLeftSpeed
```

Comment:
This won't return true unless both motor velocities are outside the desired range.